scripts/updateModHistory.php

```diff
@@ -0,0 +1,219 @@
+        if (!is_array($fileProps["contributors"])) {
+            exit("Non-array contributors list\n");
+        }
+        foreach ($fileProps["contributors"] as $contributor) {
+            $newModHistoryString .= "            \"" . $contributor . "\",\n";
+        }
+    }
+    $newModHistoryString .= "        ],\n";
+    $newModHistoryString .= "    ],\n";
+    fwrite($fp, $newModHistoryString);
+}
+fwrite($fp, "];\n");
+fclose($fp);
+
+echo "done at " . date('H:i:s') . "\n";
+
+function timeStamp(): string {
+    return "[" . date('H:i:s') . "]";
+}
+
+function processGitDiffLine($line, &$modifiedFiles): void {
+    static $currentType = "";
+    static $currentFile = "";
+
+    switch ($line) {
+        case "filename:":
+            $currentType = "filename";
+            return;
+        case "modified:":
+            $currentType = "modDateTime";
+            return;
+        case "contributors:":
+            $currentType = "contributors";
+            return;
+        case "last commit hash:":
+            $currentType = "commitHash";
+            return;
+    }
+    if ($currentType === "") {
+        return;
+    }
+
+    switch ($currentType) {
+        case "filename":
+            $currentFile = $line;
+            break;
+        case "modDateTime":
+            if ($currentFile === "") {
+                return;
+            }
+            $modifiedFiles[$currentFile]["modified"] = $line;
+            break;
+        case "contributors":
+            if ($currentFile === "") {
+                return;
+            }
+            $modifiedFiles[$currentFile]["contributors"][] = htmlspecialchars($line, ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401);
+            break;
+        case "commitHash":
+            $modifiedFiles["last commit hash"][] = $line;
+            break;
+    }
+}
+
+function verifyCommandLineOptions($commandLineOptions): void {
+    $output = timeStamp() . " - Parsing command line arguments... ";
+    echo $output;
+    if ($commandLineOptions === false
+        || !isset($commandLineOptions["docs-path"])) {
+        echo "\"--docs-path\" is a required argument\n";
+        exit(1);
+    }
+    echo "documentation path supplied\n";
+    if (isset($commandLineOptions["history-path"])) {
+        echo str_repeat(" ", strlen($output)) . "mod history file path supplied\n";
+    }
+
+    $output = timeStamp() . " - Verifying command line arguments... ";
+    echo $output;
+    if (!file_exists($commandLineOptions["docs-path"])) {
+        echo "documentation path \"" . $commandLineOptions["docs-path"] . "\" doesn't exist\n";
+        exit(1);
+    } else if (!is_dir($commandLineOptions["docs-path"])) {
+        echo "documentation path \"" . $commandLineOptions["docs-path"] . "\" is not a directory\n";
+        exit(1);
+    }
+    echo "documentation path verified\n";
+
+    if (isset($commandLineOptions["history-path"])) {
+        echo "\n" . str_repeat(" ", strlen($output)) . "mod history file path ";
```

Comment:
Possible alternative:
```suggestion
        printf("\n%*smod history file path ", strlen($output), " ");
```

Reply:
Done